### DISCUSSION TITLE
[Bug] 部分图片无法在通过域名访问时加载

### DISCUSSION THREAD

**Discussion by Corlius** [lobehub/lobehub#2730] [Self Hosting] [answered]
### 📦 部署环境

- [ ] Official
- [ ] Official Preview
- [ ] Vercel / Zeabur / Sealos
- [X] Docker
- [ ] Other

### 📌 软件版本

v0.162.11

### 💻 系统环境

- [ ] Windows
- [X] macOS
- [ ] Ubuntu
- [ ] Other Linux
- [ ] iOS
- [ ] Android
- [ ] Other

### 🌐 浏览器

- [X] Chrome
- [X] Edge
- [X] Safari
- [ ] Firefox
- [ ] Other

### 🐛 问题描述

在通过 Docker 部署 LobeChat 后，使用域名访问时，部分图片无法加载，FluentEmoji 都变成系统默认的 Emoji。然而，通过 IP 地址访问时，这些图片可以正常加载。

### 📷 复现步骤

1. 通过 Docker 部署 LobeChat。
2. 使用域名访问部署好的 LobeChat。
3. 发现部分图片无法加载，FluentEmoji 变成系统默认的 Emoji。
4. 使用 IP 地址访问部署好的 LobeChat。
5. 这些图片可以正常加载，FluentEmoji 显示正常。

### 🚦 期望结果

所有图片和 FluentEmoji 应该在使用域名和 IP 地址访问时都能正常加载。

### 📝 补充信息

使用开发者工具分析发现：

* 请求网址: `https://registry.npmmirror.com/@lobehub/assets-logo/1.2.0/files/assets/logo-3d.webp`
* 请求方法: `GET`
* 状态代码: `403 Forbidden` (来自 service worker)
* 引荐来源网址政策: `strict-origin-when-cross-origin`

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment, marked as answer]
你的域名可能被 npmmirror 黑名单了。是否是 `.top` 后缀的域名？

[a comment by an automated account is not shown]

**Comment by Corlius** [comment]
是的，是.top 域名，请问有什么解决办法吗

Arvin Xu ***@***.***>于2024年5月31日 周五01:46写道：

**Comment by arvinxx** [comment]
@Corlius 目前应该没有办法，只有换成不是 `.top` 的域名才行。

[a comment by an automated account is not shown]

**Comment by Corlius** [comment]
好的，谢谢你！

Arvin Xu ***@***.***>于2024年5月31日 周五02:13写道：

**Comment by Matinal111** [comment]
https://github.com/cnpm/unpkg-white-list/issues/54

**Comment by hbprotoss** [comment]
静态资源不能打包进lobe吗？

**Reply by arvinxx** [reply]
不能。flutemoji 全套资源 1个G

**Reply by AnotiaWang** [reply]
> 不能。flutemoji 全套资源 1个G

@arvinxx 请问是否有其它的解决方式？例如允许在环境变量里自己设置 npm mirror 服务地址，或者将 LobeChat 里用到的图标按需打包到构建产物里，这样自部署的用户可以用自己的 CDN 来加速访问

**Reply by arvinxx** [reply]
`CDN_USE_GLOBAL` 试试这个环境变量？

**Reply by AnotiaWang** [reply]
大部分图标都变成 unpkg 的了，但是首页 loading placeholder 的 logo 好像没有变，还是用的 npmmirror

https://registry.npmmirror.com/@lobehub/assets-logo/1.2.0/files/assets/logo-3d.webp

<img width="291" alt="image" src="https://github.com/user-attachments/assets/0b259158-2545-458d-abea-4e09a57541de" />

**Reply by arvinxx** [reply]
可以把没有变的列一下，然后提 issue 到 https://github.com/lobehub/lobe-ui

**Reply by wxxsfxyzm** [reply]
> `CDN_USE_GLOBAL` 试试这个环境变量？

给lobechat的docker容器加上这个环境变量似乎没有效果

**Reply by AnotiaWang** [reply]
> 给lobechat的docker容器加上这个环境变量似乎没有效果

把容器删掉重建一下试试

**Reply by AnotiaWang** [reply]
> 可以把没有变的列一下，然后提 issue 到 https://github.com/lobehub/lobe-ui

这个应该列得比较全了 https://github.com/lobehub/lobe-ui/issues/279

**Comment by datoujiejie** [comment]
用浏览器插件把Referer改一下就可以了，我用的modheader
![image](https://github.com/user-attachments/assets/3b35ea2c-2cf9-4e17-b63b-762769439367)

**Reply by Pel0915** [reply]
厉害大佬，解决了，要么依赖插件要么只能换域名
